connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/ParquetExecution.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags and
+ * the COPYRIGHT.txt file distributed with this work.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.teiid.translator.parquet;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.teiid.file.VirtualFileConnection;
+import org.teiid.language.Select;
+import org.teiid.metadata.RuntimeMetadata;
+import org.teiid.translator.DataNotAvailableException;
+import org.teiid.translator.ExecutionContext;
+import org.teiid.translator.ResultSetExecution;
+import org.teiid.translator.TranslatorException;
+
+
+public class ParquetExecution extends BaseParquetExecution implements ResultSetExecution {
+
+    private Class<?>[] expectedColumnTypes;
+
+    public ParquetExecution(Select query, ExecutionContext executionContext,
+                            RuntimeMetadata metadata, VirtualFileConnection connection, boolean immutable)
+            throws TranslatorException {
+        super(executionContext, metadata, connection, immutable);
+        this.expectedColumnTypes = query.getColumnTypes();
+        visit(query);
+    }
+
+    @Override
+    public List<?> next() throws TranslatorException, DataNotAvailableException {
+        Group row = nextRow();
+        if (row == null) {
+            return null;
+        }
+        return projectRow(row);
+    }
+
+    List<Object> projectRow(Group row) throws TranslatorException {
+        ArrayList<Object> output = new ArrayList<Object>(this.visitor.getProjectedColumns().size());
+        int fieldCount = row.getType().getFieldCount();
+        for (int field = 0; field < fieldCount; field++) {
+            Type fieldType = row.getType().getType(field);
+            int valueCount = row.getFieldRepetitionCount(field);
+            if(valueCount > 1){
```

Comment:
Yes,  I didn't think of it to be one, but it does need to be handled. I can't find any doc, so played a little. I observed that the valuecount comes out to be null for non-repeated field. In case of repeated fields, I think it is only possible with the non-annotated types.